opentracing/src/main/java/opentracing/TraceContextSource.java

```diff
@@ -1,48 +0,0 @@
-/**
- * Copyright 2016 The OpenTracing Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
- * in compliance with the License. You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software distributed under the License
- * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
- * or implied. See the License for the specific language governing permissions and limitations under
- * the License.
- */
-package opentracing;
-
-import java.util.Map;
-
-/**
- * Long-lived interface that knows how to create a root {@linkplain TraceContext} and encode/decode
- * any other.
- */
-public interface TraceContextSource {
-
-  /**
-   * Encodes or Decodes a {@linkplain TraceContext trace context} in binary or text formats.
-   */
-  TraceContextCodec codec();
-
-  /**
-   * Create a TraceContext which has no parent (and thus begins its own trace).
-   *
-   * <p>A TraceContextSource must always return the same type in successive calls to
-   * NewRootTraceContext().
-   */
-  TraceContext newRoot();
-
-  /**
-   * Creates a child context for {@code parent}, and returns both that child's own
-   * TraceContext as well as any Tags that should be added to the child's Span.
-   */
-  ChildTraceContext newChild(TraceContext parent);
-
-  interface ChildTraceContext {
```

Comment:
As I (feebly) tried to explain in https://github.com/opentracing/opentracing-java/pull/7#discussion_r50499357 there seems to be two ways to have obtained the current SpanContext.

1) You've created a brand new trace, and still adding spans to the root, are working with the SpanContext from `Tracer.newRoot()`. (I would rename this to rootContext(), since the creation function is really done via `Tracer.startTrace(..)`.

2) You're in the middle of a trace, have hold of a current Span, and are working with a `SpanContext.childContext()`. 

> In zipkin today, I don't know if any tracers propagate tags implicitly from parent to child, as a function of creating a child.

Yes, to me it looks like each SpanContext would inherit tags from any parent SpanContext, while tags added to a SpanContext do not propagate back to the parent SpanContext. A need to not inherit tags from a parent SpanContext would warrant something along the lines of `SpanContext.clear()` and `SpanContext.remove(key)`.